VehicleRental: simplify form factor mapping to allow moped rentals

flaktack opened the pull request:
### Summary

Due to differing `StreetMode` / `TraverseMode` / `RentalFormFactor` mappings `MOPED` rentals were not found.

Since both a `StreetMode → TraverseMode` and a `RentalFormFactor → TraverseMode` mapping exists, a (third) `StreetMode → RentalFormFactor` mapping is replaced with checking that the `TraverseMode` for the `StreetMode` and `RentalFormFactor` matches.

### Issue

Closes #6719

### Unit tests

A test is added for MOPED rentals.

### Documentation

No changes.

flaktack committed:
VehicleRental: standardize StreetMode - TraverseMode - FormFactor mapping

The mapping of StreetMode / TraverseMode / FormFactor is done in
multiple places and could be mapped differently.

The vehicle rental form factor logic is simplified to check if the
rental traverse mode matches the form factor's traverse mode.

This allows removing a StreetMode/RentalFormFactor mapping in
VehicleRentalEdge.

flaktack committed:
VehicleRental: add failing bicycle/moped rental test

Since the moped form factor has a BICYCLE traverse mode BIKE_RENTAL
should allow renting.

flaktack commented:
Yes, those are mopeds -- I'm not sure why its mapped to `BICYCLE` instead of `CAR` or `SCOOTER`. I can update the form factor's traverse mode if there is a consensus as to what it should be.

leonardehrenfried commented:
Looks like _I_ added the mapping. :facepalm:

leonardehrenfried commented:
To be on the safe side, I would vote to use car routing because with bike routing you're going to get all sorts of shortcuts that are not allowed on a scooter.

leonardehrenfried reviewed (dismissed):
This is a good improvement.

When someone shows up with a moped feed, we will have to decide if the current mapping to BICYCLE is still the best thing we can do.

flaktack committed:
VehicleRentalEdgeTest: clarify propulsion type

Adds a new method to avoid repeating the same parameters, while allowing
the propulsion type to be specified.

jessicaKoehnke commented on application/src/test/java/org/opentripplanner/street/model/edge/VehicleRentalEdgeTest.java:
```diff
@@ -181,23 +208,23 @@ void testBannedScooterNetworkFreeFloating() {
-  private void initEdgeAndRequest(StreetMode mode, int vehicles, int spaces) {
-    initEdgeAndRequest(mode, vehicles, spaces, false, true, true, false);
+  private void initBicycleEdgeAndRequest(int vehicles, int spaces) {
+    initEdgeAndRequest(BIKE_RENTAL, BICYCLE, HUMAN, vehicles, spaces, false, true, true, false);
```
can you put the two overloaded methods next to each other? Then it's fine for me

flaktack committed:
VehicleRentalEdgeTest: move related methods together
